lib/xrp/XrpRpc.js

```diff
@@ -0,0 +1,237 @@
+const RippleAPI = require('ripple-lib').RippleAPI;
+const promptly = require('promptly');
+const util = require('util');
+const EventEmitter = require('events');
+
+const passwordPromptAsync = util.promisify(promptly.password);
+
+class XrpRpc {
+  constructor(config) {
+    this.config = config;
+    const {
+      port,
+      host,
+      protocol,
+      address,
+    } = config;
+    const connectionString = `${protocol}://${host}:${port}`;
+    this.rpc = new RippleAPI({
+      server: connectionString,
+    });
+    this.address = address;
+    this.emitter = new EventEmitter();
+  }
+
+  async asyncCall(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc[method](args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async asyncRequest(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc.request(method, args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async unlockAndSendToAddress({ address, amount, passphrase }) {
+    if (passphrase === undefined) {
+      passphrase = await passwordPromptAsync('> ');
+    }
+    console.warn('Unlocking for a single transaction.');
+    return await this.sendToAddress({ address, amount, passphrase });
+  }
+
+  async sendToAddress({ address, amount, passphrase, tag, invoiceID, secret }, alreadyConnected = false) {
```

Comment:
I think it would be better to keep `alreadyConnected` in the first parameter. Same with the other methods.

Reply:
I agree, fixed here and everywhere else alreadyConnected is used